packages/flutter/lib/src/widgets/async.dart

```diff
@@ -693,9 +693,16 @@ class FutureBuilder<T> extends StatefulWidget {
     this.future,
     this.initialData,
     required this.builder,
+    this.rethrowError = false,
```

Comment:
This feels weird as an API, the developer can handle the error in the builder. Is this necessary?

Reply:
I agree this problem may be weird and niche. As @bernaferrari suggests, we can add this to `ThemeData.of(context).zone.something` which sounds better. wdys?

Reply:
Theme doesn't feel like the right place to put this neither.
Mayne this can be implement similar to https://api.flutter.dev/flutter/widgets/EditableText/debugDeterministicCursor.html

Reply:
This feels right. Let me know if you approve and I'll add the API.

Reply:
yes, this is the way i will go about fixing this issue. Please make sure the mechanism is only enabled during debug mode, i.e. wraps the logic in an assert